### PR TITLE
refactor: network & ipfs & logger

### PR DESCRIPTION
**Pull request by sxzz**

### CODE REVIEW

**Comment by sxzz** [comment]
I'm refactoring ES6 classes to functions. Functions are more friendly for tree-shaking. /cc @Songkeys

File: src/network.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  CBT: '0x3D1b588a6Bcd728Bb61570ced6656eA4C05e404f',
+  TIPS: '0x0058be0845952D887D1668B5545de995E12e8783',
+  MIRA: '0xAfB95CC0BD320648B3E8Df6223d9CDD05EbeDC64',
+  LINKLIST: '0xFc8C75bD5c26F50798758f387B698f207a016b6A',
+} as const satisfies Record<string, Address>
+
+export const AVAILABLE_NETWORKS = ['crossbell'] as const
+export type AvailableNetwork = (typeof AVAILABLE_NETWORKS)[number]
+
+export const IPFS_GATEWAY = 'https://w3s.link/ipfs/'
```
Comment (Songkeys) [inline src/network.ts:16]: This makes it impossible to change the gateway for user

File: src/network.ts [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+export const JSON_RPC_ADDRESS =
+  // @ts-ignore
+  globalThis.process?.env.CROSSBELL_RPC_ADDRESS ?? 'https://rpc.crossbell.io'
```
Comment (Songkeys) [inline src/network.ts:19]: This makes it impossible to change the gateway for *frontend* user

File: src/network.ts [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+export const CROSSBELL_CHAIN = {
```
[the remark this answers is not in the document]
Comment (sxzz) [reply src/network.ts:21]: Great. I didn't know that viem had already included crossbell chain.

**Pull request merged**